### PR TITLE
Prerelease fixup

### PR DESCRIPTION
**Pull request by komidore64**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.20
* [ ] 1.19
* [ ] 1.18

See Foreman's [plugin maintainer documentation](https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

File: packages/foreman/rubygem-hammer_cli/rubygem-hammer_cli.spec [hunk 1]
```diff
@@ -14,11 +14,11 @@
 Summary: Universal command-line interface for Foreman
 Name: %{?scl_prefix}rubygem-%{gem_name}
 Version: 0.15
```
Comment (ekohl) [inline packages/foreman/rubygem-hammer_cli/rubygem-hammer_cli.spec]: Mind bumping this to 0.16 while you're at it to match https://github.com/theforeman/hammer-cli/commit/298d6077c24a340e0c135c3e8af3097e82ccfc8a

File: packages/foreman/rubygem-hammer_cli_foreman/rubygem-hammer_cli_foreman.spec [hunk 2]
```diff
@@ -4,19 +4,19 @@
...
-%global release 2
-%global prerelease .pre.develop
+%global release 3
+%global prerelease pre.develop
 
 %{!?_root_sysconfdir:%global _root_sysconfdir %{_sysconfdir}}
 
 Summary: Universal command-line interface for Foreman
 Name: %{?scl_prefix}rubygem-%{gem_name}
 Version: 0.15
```
Comment (ekohl) [inline packages/foreman/rubygem-hammer_cli_foreman/rubygem-hammer_cli_foreman.spec]: Current git master is also 0.16 which we should match: https://github.com/theforeman/hammer-cli-foreman/commit/fd102084e3a562e1e437568412b2de3c3f51561e

File: packages/katello/katello-installer-base/katello-installer-base.spec [hunk 3]
```diff
@@ -65,7 +65,7 @@ A set of tools for installation of Katello and Katello Capsule,
 including Foreman and Foreman Proxy.
 
 %prep
-%setup -q -n katello-installer-%{version}%{?prerelease}
+%setup -q -n katello-installer-%{version}%{?prerelease:-}%{?prerelease}
```
Comment (ekohl) [inline packages/katello/katello-installer-base/katello-installer-base.spec]: I do wonder if RPM isn't able to derive this itself from the Source0.

File: packages/katello/rubygem-hammer_cli_katello/rubygem-hammer_cli_katello.spec [hunk 4]
```diff
@@ -6,17 +6,17 @@
...
 %global confdir hammer
 
-%global release 3
-%global prerelease .pre.master
+%global release 4
+%global prerelease pre.master
 
 Summary: Katello command plugin for the Hammer CLI
 Name:    %{?scl_prefix}rubygem-%{gem_name}
 Version: 0.15
```
Comment (ekohl) [inline packages/katello/rubygem-hammer_cli_katello/rubygem-hammer_cli_katello.spec]: Also moved to 0.16: https://github.com/Katello/hammer-cli-katello/commit/f1a3f73d940b6c3cc604af2bb640842a61a07f79

**Comment by komidore64** [comment]
@ekohl yeah I'll take care of those bumps.

**Comment by komidore64** [comment]
@ekohl please keep in mind that `hammer_cli`, `hammer_cli_foreman`, nor `hammer_cli_katello` have yet to push any such  `0.16` to rubygems.org yet, so building will probably fail.

File: packages/foreman/rubygem-hammer_cli/rubygem-hammer_cli.spec [hunk 5]
```diff
@@ -127,6 +127,12 @@ install -m 755 .%{gem_instdir}/config/cli_config.template.yml \
 %{gem_instdir}/test
 
 %changelog
+* Thu Oct 25 2018 Adam Price <komidore64@gmail.com> - 0.16-0.1.pre.develop
+- bump version to 0.16
+
+* Wed Oct 24 2018 Adam Price <komidore64@gmail.com> - 0.15-0.3.pre.develop
```
Comment (ekohl) [inline packages/foreman/rubygem-hammer_cli/rubygem-hammer_cli.spec]: You could leave out this version and just keep it to a single entry

File: packages/katello/rubygem-katello/rubygem-katello.spec [hunk 6]
```diff
@@ -359,7 +359,11 @@ rm -f %{buildroot}%{foreman_webpack_plugin}/*.js.map
 %{gem_instdir}/webpack
 
 %changelog
-* Thu Oct 18 2018 Eric D. Helms <ericdhelms@gmail.com> - 3.10.0-1
+* Wed Oct 24 2018 Adam Price <komidore64@gmail.com> - 3.10.0-0.2.master
+- breakout period and hyphen from prerelease macro
+- set prerelease to master
+
+* Thu Oct 18 2018 Eric D. Helms <ericdhelms@gmail.com> - 3.10.0-0.1.master
```
Comment (ekohl) [inline packages/katello/rubygem-katello/rubygem-katello.spec]: Here you could also remove the master to keep Eric's changelog in tact

File: packages/katello/rubygem-katello/rubygem-katello.spec [hunk 7]
```diff
@@ -4,19 +4,19 @@
 %global foreman_min_version 1.18.0
 %global plugin_name katello
 %global gem_name katello
-# %%global prerelease .rc1
+%global prerelease master
```
[the remark this answers is not in the document]
Comment (komidore64) [reply packages/katello/rubygem-katello/rubygem-katello.spec]: oh whoops! you're right, i missed this one

File: packages/katello/rubygem-katello/rubygem-katello.spec [hunk 8]
```diff
@@ -359,6 +359,10 @@ rm -f %{buildroot}%{foreman_webpack_plugin}/*.js.map
 %{gem_instdir}/webpack
 
 %changelog
+* Wed Oct 24 2018 Adam Price <komidore64@gmail.com> - 3.10.0-0.2.master
```
Comment (ehelms) [inline packages/katello/rubygem-katello/rubygem-katello.spec]: This is just missing `pre` in it and then its good to go.

**Pull request merged**